### PR TITLE
add enable_tcp_reset to azurerm_lb_rule & azurerm_lb_nat_rule

### LINKED ISSUE #3404: Add enableTcpReset parameter for Standard Load Balancer rules and nat rules
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The Standard LB introduces support for [TCP Reset](https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-tcp-reset)

The outbound rule already exposes the functionality : [enable_tcp_reset](https://www.terraform.io/docs/providers/azurerm/r/loadbalancer_outbound_rule.html#enable_tcp_reset)
 
However it's missing from [loadbalancer rule](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_loadbalancer_rule.go) and [loadbalancer nat rule](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_loadbalancer_nat_rule.go)

### New or Affected Resource(s)

* `azurerm_lb_nat_rule`
* `azurerm_lb_rule`

### Potential Terraform Configuration

see https://github.com/terraform-providers/terraform-provider-azurerm/blob/54ce52395ce71bcde2e0e983e06061159faea106/azurerm/resource_arm_loadbalancer_outbound_rule.go#L79-L83

### Note 
Outbound rules is for `Standard` sku only so `enableTcpReset` is always allowed as a parameter.
For `lb_rule` and `nat_rule`, it needs to validate that the lb sku is `Standard`.

### PR DESCRIPTION
**Pull request by katbyte**
fixes #3404

also adds `idle_timeout_in_minutes` to`azurerm_lb_nat_rule `

### CODE REVIEW

File: azurerm/internal/services/network/resource_arm_loadbalancer_nat_rule.go [hunk 1]
```diff
@@ -93,6 +87,24 @@ func resourceArmLoadBalancerNatRule() *schema.Resource {
...
+
+			"enable_tcp_reset": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"idle_timeout_in_minutes": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/network/resource_arm_loadbalancer_nat_rule.go]: why is this computed?

File: azurerm/internal/services/network/tests/resource_arm_loadbalancer_nat_rule_test.go [hunk 2]
```diff
@@ -290,7 +301,47 @@ func testCheckAzureRMLoadBalancerNatRuleDisappears(natRuleName string, lb *netwo
...
+
+resource "azurerm_lb_nat_rule" "test" {
+  location                       = "${azurerm_resource_group.test.location}"
+  resource_group_name            = "${azurerm_resource_group.test.name}"
+  loadbalancer_id                = "${azurerm_lb.test.id}"
+  name                           = "%[4]s"
+  protocol                       = "Tcp"
+  frontend_port                  = 3389
+  backend_port                   = 3389
+  frontend_ip_configuration_name = "one-%[1]d"
```
Comment (tombuildsstuff) [inline azurerm/internal/services/network/tests/resource_arm_loadbalancer_nat_rule_test.go]:
```suggestion
  frontend_ip_configuration_name = azurerm_lb.frontend_ip_configuration.0.name
```

File: azurerm/internal/services/network/tests/resource_arm_loadbalancer_rule_test.go [hunk 3]
```diff
@@ -391,58 +432,64 @@ resource "azurerm_lb_rule" "test" {
   location                       = "${azurerm_resource_group.test.location}"
   resource_group_name            = "${azurerm_resource_group.test.name}"
   loadbalancer_id                = "${azurerm_lb.test.id}"
-  name                           = "%s"
+  name                           = "%[4]s"
   protocol                       = "Tcp"
   frontend_port                  = 3389
   backend_port                   = 3389
-  frontend_ip_configuration_name = "one-%d"
+  frontend_ip_configuration_name = "one-%[1]d"
```
Comment (tombuildsstuff) [inline azurerm/internal/services/network/tests/resource_arm_loadbalancer_rule_test.go]:
```suggestion
  frontend_ip_configuration_name = azurerm_lb.frontend_ip_configuration.0.name
```

File: azurerm/internal/services/network/tests/resource_arm_loadbalancer_rule_test.go [hunk 4]
```diff
@@ -391,58 +432,64 @@ resource "azurerm_lb_rule" "test" {
...
   location            = "${azurerm_resource_group.test.location}"
   resource_group_name = "${azurerm_resource_group.test.name}"
   sku                 = "Standard"
 
   frontend_ip_configuration {
-    name                 = "one-%d"
+    name                 = "one-%[1]d"
     public_ip_address_id = "${azurerm_public_ip.test.id}"
   }
 }
```
Comment (tombuildsstuff) [inline azurerm/internal/services/network/tests/resource_arm_loadbalancer_rule_test.go]: we likely could pull all of this out into ` _template` helper

File: website/docs/r/loadbalancer_nat_rule.html.markdown [hunk 5]
```diff
@@ -62,7 +62,9 @@ The following arguments are supported:
 * `protocol` - (Required) The transport protocol for the external endpoint. Possible values are `Udp`, `Tcp` or `All`.
 * `frontend_port` - (Required) The port for the external endpoint. Port numbers for each Rule must be unique within the Load Balancer. Possible values range between 1 and 65534, inclusive.
 * `backend_port` - (Required) The port used for internal connections on the endpoint. Possible values range between 1 and 65535, inclusive.
+* `idle_timeout_in_minutes` - (Optional) Specifies the timeout for the Tcp idle connection. The value can be set between 4 and 30 minutes. The default value is 4 minutes. This element is only used when the protocol is set to Tcp.
 * `enable_floating_ip` - (Optional) Enables the Floating IP Capacity, required to configure a SQL AlwaysOn Availability Group.
+* `enable_tcp_reset` - (Optional) Indicates whether tcp reset is disabled or enabled. Default `false`.
```
Comment (tombuildsstuff) [inline website/docs/r/loadbalancer_nat_rule.html.markdown]: we can make this:

```suggestion
* `enable_tcp_reset` - (Optional) Is TCP Reset Enabled for this Load Balancer? Defaults to `false`.
```

File: website/docs/r/loadbalancer_nat_rule.html.markdown [hunk 6]
```diff
@@ -62,7 +62,9 @@ The following arguments are supported:
 * `protocol` - (Required) The transport protocol for the external endpoint. Possible values are `Udp`, `Tcp` or `All`.
 * `frontend_port` - (Required) The port for the external endpoint. Port numbers for each Rule must be unique within the Load Balancer. Possible values range between 1 and 65534, inclusive.
 * `backend_port` - (Required) The port used for internal connections on the endpoint. Possible values range between 1 and 65535, inclusive.
+* `idle_timeout_in_minutes` - (Optional) Specifies the timeout for the Tcp idle connection. The value can be set between 4 and 30 minutes. The default value is 4 minutes. This element is only used when the protocol is set to Tcp.
```
Comment (tombuildsstuff) [inline website/docs/r/loadbalancer_nat_rule.html.markdown]:
```suggestion
* `idle_timeout_in_minutes` - (Optional) Specifies the idle timeout in minutes for TCP connections. Valid values are between `4` and `30` minutes. Defaults to `4` minutes.
```

File: website/docs/r/loadbalancer_rule.html.markdown [hunk 7]
```diff
@@ -66,6 +66,7 @@ The following arguments are supported:
 * `idle_timeout_in_minutes` - (Optional) Specifies the timeout for the Tcp idle connection. The value can be set between 4 and 30 minutes. The default value is 4 minutes. This element is only used when the protocol is set to Tcp.
 * `load_distribution` - (Optional) Specifies the load balancing distribution type to be used by the Load Balancer. Possible values are: `Default` – The load balancer is configured to use a 5 tuple hash to map traffic to available servers. `SourceIP` – The load balancer is configured to use a 2 tuple hash to map traffic to available servers. `SourceIPProtocol` – The load balancer is configured to use a 3 tuple hash to map traffic to available servers. Also known as Session Persistence, where  the options are called `None`, `Client IP` and `Client IP and Protocol` respectively.
 * `disable_outbound_snat` - (Optional) Indicates whether outbound snat is disabled or enabled. Default false.
+* `enable_tcp_reset` - (Optional) Indicates whether tcp reset is disabled or enabled. Default `false`.
```
Comment (tombuildsstuff) [inline website/docs/r/loadbalancer_rule.html.markdown]:
```suggestion
* `enable_tcp_reset` - (Optional) Is TCP Reset Enabled for this Load Balancer? Defaults to `false`.
```

File: azurerm/internal/services/network/resource_arm_loadbalancer_nat_rule.go (hunk 1 above)
Comment (katbyte) [reply azurerm/internal/services/network/resource_arm_loadbalancer_nat_rule.go]: It's the way lb rule is and allows the API to set the default: 
```
idle_timeout_in_minutes:        "4" => ""
```

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2020-01-13 at 17 00 03

**Pull request merged**